### PR TITLE
Fix Total Rows to remove erroneous fact join

### PR DESCRIPTION
**Pull request by ryankwagner**

### CODE REVIEW

**Comment by patelh** [comment]
Hard to see what is being fixed without any tests...

**Comment by ryankwagner** [comment]
totalRows was returning a fact join on non-pk dim filters.  This removes the fact join that it was forcing in RequestModel L615.  I'll add in a test to demonstrate the functionality.

File: service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala [hunk 1]
```diff
@@ -257,6 +257,7 @@ case class RowCountCurator(protected val requestModelValidator: CuratorRequestMo
                 //remove all fields except primary key
                 //remove all sorts
                 val primaryKeyAliasFields = sourcePipeline.bestDimCandidates.map(dim => Field(dim.publicDim.primaryKeyByAlias, None, None)).toIndexedSeq
+                val primaryFilteredExpressions = sourcePipeline.requestModel.reportingRequest.filterExpressions.filter(filter => primaryKeyAliasFields.contains(Field(filter.field, None, None)))
```
Comment (patelh) [inline service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: I don't think you want to remove all non-pk filters here.  Doesn't make sense given you want to have total rows reflect the number of rows with all filters applied.

Comment (ryankwagner) [reply service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: This is true.  I am considering other ways to approach this issue.

Comment (patelh) [reply service/src/main/scala/com/yahoo/maha/service/curators/Curator.scala]: Try seeing why the unexpected join is constructed in request model.

**Comment by ryankwagner** [comment]
Removing this PR, as the removal of filtering is more than what we want.

**Pull request closed without merging**